oxidation/libparsec/crates/types/src/certif.rs

```diff
@@ -41,33 +62,83 @@ fn check_author_allow_root(
             CertificateSignerOwned::User(author_id),
             CertificateSignerRef::User(expected_author_id),
         ) if author_id != expected_author_id => {
-            return Err(Box::new(DataError::UnexpectedAuthor {
-                expected: expected_author_id.clone(),
-                got: Some(author_id.clone()),
-            }))
+            return Err(DataError::UnexpectedAuthor {
+                expected: Box::new(expected_author_id.clone()),
+                got: Some(Box::new(author_id.clone())),
+            })
         }
         (CertificateSignerOwned::Root, CertificateSignerRef::User(expected_author_id)) => {
-            return Err(Box::new(DataError::UnexpectedAuthor {
-                expected: expected_author_id.clone(),
+            return Err(DataError::UnexpectedAuthor {
+                expected: Box::new(expected_author_id.clone()),
                 got: None,
-            }))
+            })
         }
         _ => (),
     }
 
     Ok(())
 }
 
+pub enum UnsecureSkipValidationReason {
+    // Certificate must have been validated prior to being added to the local storage,
+    // on top of that the certificate are store encrypted so they cannot be tempered.
+    // Hence it's safe not to validate the certificate when reading thom from the
+    // local storage.
+    DataFromLocalStorage,
```

Comment:
```suggestion
    /// Certificate must have been validated prior to being added to the local storage.
    /// On top of that the certificates are stored encrypted so they cannot be tempered.
    /// Hence it's safe not to validate the certificate when reading them from the
    /// local storage.
    DataFromLocalStorage,
```

Reply:
> On top of that the certificates are stored encrypted so they cannot be tempered.

Is it because the encrypted data if tempered will fail the decryption ?